### PR TITLE
feat(server): remove server context

### PR DESCRIPTION
**Pull request by archbear**
## Summary by CodeRabbit

- **New Features**
	- Enhanced modularity in command functions by decoupling server context, utilizing the client package for configuration access.
	- Improved logging mechanism through consistent wrapping of the logger in command initialization.

- **Bug Fixes**
	- Streamlined command setup process by eliminating unnecessary server context initialization, simplifying the execution flow.

- **Refactor**
	- Removed `Context` struct for server management, directly interacting with `Viper` for configuration, improving code clarity.
	- Adjusted parameter handling in command functions to enhance modularity and maintainability.

- **Chores**
	- Cleaned up unnecessary import statements and commented out unused type definitions for a leaner codebase.

### CODE REVIEW

File: mod/consensus/pkg/cometbft/service/server/start.go [hunk 1]
```diff
@@ -151,17 +156,19 @@ is performed. Note, when enabled, gRPC will also be automatically enabled.
 }
 
 func start[T types.Application](
-	svrCtx *Context,
+	logger log.Logger,
```
Comment (ocnc2) [inline mod/consensus/pkg/cometbft/service/server/start.go:159]: huge

**Review by ocnc2 — APPROVED** [review]
big if true

File: mod/cli/pkg/commands/types.go [hunk 2]
```diff
@@ -21,20 +21,14 @@
...
-// PostSetupFn is a function that is called after the application is created
-// and the cosmos server is started.
-type PostSetupFn[T types.Node] func(
-	app T, svrCtx *server.Context, clientCtx client.Context,
-	ctx context.Context, g *errgroup.Group) error
+// // PostSetupFn is a function that is called after the application is created
+// // and the cosmos server is started.
+// type PostSetupFn[T types.Node] func(
+// 	app T, svrCtx *server.Context, clientCtx client.Context,
+// 	ctx context.Context, g *errgroup.Group) error
```
Comment (ocnc) [inline mod/cli/pkg/commands/types.go:31]:
```suggestion
```

File: mod/consensus/pkg/cometbft/service/server/cmd/execute.go [hunk 3]
```diff
@@ -60,9 +59,9 @@ func Execute(rootCmd *cobra.Command, envPrefix, defaultHome string) error {
 // CreateExecuteContext returns a base Context with server and client context
 // values initialized.
 func CreateExecuteContext(ctx context.Context) context.Context {
-	srvCtx := server.NewDefaultContext()
+	// srvCtx := server.NewDefaultContext()
```
Comment (ocnc) [inline mod/consensus/pkg/cometbft/service/server/cmd/execute.go:62]:
```suggestion
```

File: mod/consensus/pkg/cometbft/service/server/cmd/execute.go [hunk 4]
```diff
@@ -60,9 +59,9 @@ func Execute(rootCmd *cobra.Command, envPrefix, defaultHome string) error {
...
 	ctx = context.WithValue(ctx, client.ClientContextKey, &client.Context{})
-	ctx = context.WithValue(ctx, server.ServerContextKey, srvCtx)
+	// ctx = context.WithValue(ctx, server.ServerContextKey, srvCtx)
```
Comment (ocnc) [inline mod/consensus/pkg/cometbft/service/server/cmd/execute.go:64]:
```suggestion
```

**Review by itsdevbear — APPROVED** [review]

**Pull request merged**